src/components/app/App.js

```diff
@@ -0,0 +1,343 @@
+import AboutTheAuthor from '../aboutTheAuthor/AboutTheAuthor';
+import Footer from '../footer/Footer';
+import SignUpPopup from '../popup/SignUpPopup';
+import CurrentUserContext from '../../contexts/CurrentUserContext';
+import * as auth from '../../utils/auth';
+import LoginPopup from '../popup/LoginPopup';
+import Main from '../main/Main';
+import ProtectedRoute from '../protectedRoute/ProtectedRoute';
+import SavedArticles from '../savedArticles/SavedArticles';
+import SearchBar from '../searchBar/SearchBar';
+import NotFound from '../notFound/NotFound';
+
+function App() {
+  // eslint-disable-next-line
+  const currentUserContext = React.useContext(CurrentUserContext);
+  const safeDocument = typeof document !== 'undefined' ? document : {};
+  const html = safeDocument.documentElement;
+  const history = useHistory();
+  const [currentUser, setCurrentUser] = React.useState();
+  const [articlesArray, setArticlesArray] = React.useState([]);
+  const [showLessArray, setShowLessArray] = React.useState([]);
+  const [savedArticles, setSavedArticles] = React.useState([]);
+  const [isHomePage, setIsHomePage] = React.useState(true);
+  const [loggedIn, setLoggedIn] = React.useState(false);
+  const [isResultsOpen, setIsResultsOpen] = React.useState(false);
+  const [isSignUpPopupOpen, setIsSignUpPopupOpen] = React.useState();
+  const [isLoginPopupOpen, setIsLoginPopupOpen] = React.useState();
+  const [isPreloader, setIsPreloader] = React.useState(true);
+  const [isNotFound, setIsNotFound] = React.useState(false);
+  const [q, setQ] = React.useState('news');
+
+  const handleSignUpClick = () => setIsSignUpPopupOpen(true);
+
+  const handleLoginClick = () => setIsLoginPopupOpen(true);
+
+  const closeAllPopups = () => {
+    setIsSignUpPopupOpen(false);
+    setIsLoginPopupOpen(false);
+    if (window.innerWidth < 520) {
+      noScroll();
+    }
+    history.push('/');
+  };
+
+  const toggleNoScroll = () => html.classList.toggle('no-scroll');
+
+  const noScroll = () => html.classList.add('no-scroll');
+
+  // ! Header handling
+  // * Handling the logout click
+  const handleLogout = () => {
+    setLoggedIn(false);
+    setIsHomePage(true);
+    setSavedArticles([]);
+    localStorage.removeItem('jwt');
+    history.push("/");
+  };
+
+  // * getting articles
+  React.useEffect(() => {
+    const closeByEscape = (evt) => {
+      if (evt.key === 'Escape') {
+        closeAllPopups();
+      }
+    };
+
+    document.addEventListener('keydown', closeByEscape);
+    return () => document.removeEventListener('keydown', closeByEscape);
+    // eslint-disable-next-line
+  }, []);
+
+  // ! Adding event listener for the page
+  // ! Mouse event
+  React.useEffect(() => {
+    const closeByClick = (evt) => {
+      if (evt.target.classList.contains("popup")) {
+        closeAllPopups();
+      }
+    }
+
+    document.addEventListener('mouseup', closeByClick);
+    return () => document.removeEventListener('mouseup', closeByClick);
+  });
```

Comment:
you need to add `  }, []);` here (an empty dependencies array) like in line 76 to prevent constant useEffect calling when rerendering


```js


  }, []);
```